packages/hub/src/utils/checkCredentials.ts

```diff
@@ -0,0 +1,11 @@
+import type { Credentials } from "../types";
+
+export function checkCredentials(credentials?: Credentials): void {
+	if (!credentials || credentials.accessToken === undefined || credentials.accessToken === null) {
+		return;
```

Comment:
it's legal to not provide access token, it will just be rate limited - so it's not an error if token is not provided. Is mu understanding right?

Reply:
For some `@huggingface/hub` functions, like listing files in a repo, the token is optional. It can be useful if the repo is private, otherwise it's not needed.